cfme/cloud/instance/__init__.py

```diff
@@ -101,25 +96,60 @@ def __init__(self, name, provider, template_name=None, appliance=None):
         Navigatable.__init__(self, appliance=appliance)
 
-    def create(self):
+    def create(self, form_values, cancel=False):
         """Provisions an instance with the given properties through CFME
+
+        Args:
+            form_values: dictionary of form values for provisioning, structured into tabs
+
+        Note:
+            Calling create on a sub-class of instance will generate the properly formatted
+            dictionary when the correct fields are supplied.
         """
-        raise NotImplementedError('create is not implemented.')
-
-    def load_details(self, refresh=False):
-        navigate_to(self, 'Details')
-        # BZ 1389299
-        # TODO: add refresh support back
-
-    def get_detail(self, *args, **kwargs):
-        # TODO: remove when bug 1389299 is fixed to use common.vm.get_detail()
-        # Navigate to all first to force reload of details screen without using refresh
-        navigate_to(self, 'All')
-        self.load_details()
-        if kwargs.get('icon_href', False):
-            return InfoBlock.icon_href(*kwargs.get('properties'))
+        view = navigate_to(self, 'Provision')
+
+        # Only support 1 security group for now
+        # TODO: handle multiple
+        if 'environment' in form_values and 'security_groups' in form_values['environment'] and \
+                isinstance(form_values['environment']['security_groups'], (list, tuple)):
+
+            first_group = form_values['environment']['security_groups'][0]
+            recursive_update(form_values, {'environment': {'security_groups': first_group}})
+
+        view.form.fill(form_values)
+
+        if cancel:
+            view.form.cancel_button.click()
+            # Redirects to Instance All, but flash is defined in BaseLoggedInPage so use directly
+            view = self.browser.create_view(BaseLoggedInPage)
+            wait_for(func=view.flash.assert_success_message, func_arg=self.PROVISION_CANCEL,
+                     fail_condition=[], timeout=10, delay=2, message='wait for flash success')
+        else:
+            view.form.submit_button.click()
+            # TODO this redirects to service.request, create_view on it when it exists for flash
+            wait_for(flash.get_messages, fail_condition=[], timeout=10, delay=2,
+                     message='wait for Flash Success')
+            flash.assert_success_message(self.PROVISION_START)
+
+    def update(self, values, cancel=False, reset=False):
+        """Update cloud instance
+
+        Args:
+            values: Dictionary of form key/value pairs
+            cancel: Boolean, cancel the form submission
+            reset Boolean, reset form after fill - returns immediately after reset
+        Note:
+            The edit form contains a 'Reset' button - if this is c
+        """
+        view = navigate_to(self, 'Edit')
+        # form is the view's parent
+        view.form.fill(values)
+        if reset:
+            view.form.reset_button.click()
```

Comment:
I would think in the case where the caller wants the form reset after fill, they may be testing reset and this should return immediately?

I did add utils.Updateable to the Instance class